Git-commit: c024f6f11d4d2d9b4fa2be2ba4bc01a5bbe6ee4b
From: Sathish Narasimman <sathish.narasimman@intel.com>
Date: Mon, 5 Apr 2021 20:00:41 +0530
Subject: [PATCH] Bluetooth: LL privacy allow RPA
References: bsn#19
Patch-mainline: v5.10.121

[ Upstream commit 8ce85ada0a05e21a5386ba5c417c52ab00fcd0d1 ]

allow RPA to add bd address to whitelist

Signed-off-by: Sathish Narasimman <sathish.narasimman@intel.com>
Signed-off-by: Marcel Holtmann <marcel@holtmann.org>
Signed-off-by: Sasha Levin <sashal@kernel.org>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 net/bluetooth/hci_request.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/bluetooth/hci_request.c b/net/bluetooth/hci_request.c
index 2405e1ffebbd..eb4c1c18eb01 100644
--- a/net/bluetooth/hci_request.c
+++ b/net/bluetooth/hci_request.c
@@ -842,6 +842,10 @@ static u8 update_white_list(struct hci_request *req)
 	 */
 	bool allow_rpa = hdev->suspended;
 
+	if (use_ll_privacy(hdev) &&
+	    hci_dev_test_flag(hdev, HCI_ENABLE_LL_PRIVACY))
+		allow_rpa = true;
+
 	/* Go through the current white list programmed into the
 	 * controller one by one and check if that address is still
 	 * in the list of pending connections or list of devices to
-- 
2.35.1

